File: typescript/prototype/underline/Underline.ts
export class Underline {
  private decorator: string;

  constructor(_decorator: string) {
    this.decorator = _decorator;
  }

  public use = (msg: string): string => {
    let line = "";
    for (let i = 0; i < msg.length; i++) line += this.decorator;

    return `\n${msg}\n${line}\n`;
  };
}

File: go/prototype/underline/underline.go
package underline

type Underline struct {
	decorator string
}

func New(deco string) *Underline {
	return &Underline{deco}
}

func (u *Underline) Use(msg string) string {
	var line string

	length := len(msg)
	for i := 0; i < length; i++ {
		line += u.decorator
	}

	result := "\n" +
		msg + "\n" +
		line + "\n"

	return result
}
